sessiontxn/isolation/readcommitted.go

```diff
@@ -178,6 +183,30 @@ func (p *PessimisticRCTxnContextProvider) getStmtTS() (ts uint64, err error) {
 	return
 }
 
+func (p *PessimisticRCTxnContextProvider) getStmtForUpdateTS() (ts uint64, err error) {
+	if p.stmtTS != 0 {
+		return p.stmtTS, nil
+	}
+
+	var txn kv.Transaction
+	if txn, err = p.ActivateTxn(); err != nil {
+		return 0, err
+	}
+
+	p.prepareStmtTS()
+	if ts, err = p.stmtTSFuture.Wait(); err != nil {
+		return 0, err
+	}
+
+	txn.SetOption(kv.SnapshotTS, ts)
+	p.stmtTS = ts
+
+	if p.checkTSInWriteStmt {
+		txn.SetOption(kv.IsolationLevel, kv.RCCheckTS)
+	}
```

Comment:
I think you should add some comments to explain why this is needed.

By the way this looks like the only difference between `getStmtTS`? Maybe you can reuse the former part of the code like

```go
	ts, err = p.getStmtTS()
	if p.checkTSInWriteStmt {
		txn.SetOption(kv.IsolationLevel, kv.RCCheckTS)
	}
	return
```

Reply:
> > The transaction can't turn into RCCheckTS mode, but this sql will.
> 
> I find `kv.IsolationLevel` of the active txn is changed:
> 
> ```go
> if p.checkTSInWriteStmt {
> 	txn.SetOption(kv.IsolationLevel, kv.RCCheckTS)
> }
> ```
> 
> Maybe I'm missing something... Where is it reset for the next statement?

Oh, I forget to reset it.

Reply:
done